realm/src/main/java/io/realm/Realm.java

```diff
@@ -225,9 +226,16 @@ protected void finalize() throws Throwable {
      * <p>
      * It's important to always remember to close Realm instances when you're done with it in order
      * not to leak memory, file descriptors or grow the size of Realm file out of measure.
+     *
+     * @throws java.lang.IllegalStateException if trying to close Realm on a different thread than the
+     * one it was created on.
      */
     @Override
     public void close() {
+        if (this.threadId != Thread.currentThread().getId()) {
```

Comment:
This logic is now used in two places. It might as well be a private method taking a string as a parameter IMO